doc/apiv3-documentation.apib

```diff
@@ -2271,6 +2271,397 @@ Lists the columns available for work packages of this project in the form of fie
+|:-------------------:| -------------------------------------------------------------------- | --------------------------------------------- |
+| update              | Updates the relation between the two work packages                   | **Permission**: manage work package relations |
+| delete              | Destroys the relation between the two work packages                  | **Permission**: manage work package relations |
+
+## Linked Properties
+| Link          | Description                           | Type          | Constraints | Supported operations | Condition                                     |
+|:-------------:|-------------------------------------- | ------------- | ----------- | -------------------- | --------------------------------------------- |
+| self          | This relation                         | Relation      | not null    | READ                 | **Permission**: view work packages            |
+| from          | The emanating work package            | WorkPackage   | not null    | READ                 | **Permission**: view work packages            |
+| to            | The work package the relation ends in | WorkPackage   | not null    | READ                 | **Permission**: view work packages            |
+
+## Local Properties
+| Property         | Description                                                   | Type    | Constraints                                                                                                   | Supported operations |
+| :---------------:| ------------------------------------------------------------- | ------- | ------------------------------------------------------------------------------------------------------------- | -------------------- |
+| id               | Relation ID                                                   | Integer | x > 0                                                                                                         | READ                 |
+| type             | Which kind of relation (blocks, precedes, etc.)               | String  | in: relates, duplicates, duplicated, blocks, blocked, precedes, follows, includes, partof, requires, required | READ / WRITE         |
+| reverseType      | The kind of relation from the other WP's perspective          | String  | in: relates, duplicates, duplicated, blocks, blocked, precedes, follows, includes, partof, requires, required | READ                 |
+| description      | Short text further describing the relation                    | String  |                                                                                                               | READ / WRITE         |
+| delay            | The delay in days between closing of `from` and start of `to` | Integer | x >= 0                                                                                                        | READ / WRITE         |
```

Comment:
I believe that the types of relations will more and more evolve into distinct resources with their own attributes. One can already see the beginning of it in the `delay` attribute which is unique to `follows`/`precedes` relations. As such, the currently implemented subtypes for relations seem to be a sensible approach to me.